src/client/components/localisation/network.ts

```diff
@@ -22,27 +27,49 @@ export class LocalisationNetwork {
 
   @action
   private onSensors = (sensors: Sensors) => {
-    // TODO (Annable): Remove hardcoded values.
+    // TODO (Annable): Support multiple robots.
     const robot = this.model.robots[0]
-    robot.motors.rightShoulderPitch.angle = Number(sensors.servo[0].presentPosition)
-    robot.motors.leftShoulderPitch.angle = Number(sensors.servo[1].presentPosition)
-    robot.motors.rightShoulderRoll.angle = Number(sensors.servo[2].presentPosition)
-    robot.motors.leftShoulderRoll.angle = Number(sensors.servo[3].presentPosition)
-    robot.motors.rightElbow.angle = Number(sensors.servo[4].presentPosition)
-    robot.motors.leftElbow.angle = Number(sensors.servo[5].presentPosition)
-    robot.motors.rightHipYaw.angle = Number(sensors.servo[6].presentPosition)
-    robot.motors.leftHipYaw.angle = Number(sensors.servo[7].presentPosition)
-    robot.motors.rightHipRoll.angle = Number(sensors.servo[8].presentPosition)
-    robot.motors.leftHipRoll.angle = Number(sensors.servo[9].presentPosition)
-    robot.motors.rightHipPitch.angle = Number(sensors.servo[10].presentPosition)
-    robot.motors.leftHipPitch.angle = Number(sensors.servo[11].presentPosition)
-    robot.motors.rightKnee.angle = Number(sensors.servo[12].presentPosition)
-    robot.motors.leftKnee.angle = Number(sensors.servo[13].presentPosition)
-    robot.motors.rightAnklePitch.angle = Number(sensors.servo[14].presentPosition)
-    robot.motors.leftAnklePitch.angle = Number(sensors.servo[15].presentPosition)
-    robot.motors.rightAnkleRoll.angle = Number(sensors.servo[16].presentPosition)
-    robot.motors.leftAnkleRoll.angle = Number(sensors.servo[17].presentPosition)
-    robot.motors.headPan.angle = Number(sensors.servo[18].presentPosition)
-    robot.motors.headTilt.angle = Number(sensors.servo[19].presentPosition)
+
+    const { translation, rotation } = decompose(fromProtoMat44(sensors.world!))
+    robot.position.set(translation.x, translation.z, translation.y)
+    robot.heading = new Euler().setFromQuaternion(rotation).z
```

Comment:
Should put the whole quaternion on the model as it's all used. Also robot.heading could be confusing as localisation will likely want to use a similar name. Ideally we should start using the same naming convention we are using on the robots. So that would make this Rtw or Rwt depending on if it was transposed.